### PR TITLE
KFSPTS-17854 upgrade to the 1/09/20, 1/23/20, 1/30/2020 version of financials

### PR DESCRIPTION
**Pull request by jhulslander**
Please note there are related PRs in cu-kfs-upgrades and nonprodSQL.  All three should be merged together.  Please take special review of SalaryExpenseTransferDocument.java.  I believe base code had become up to date with our overlay for this class, so I removed it.  I may have missed something, so I would like a second set of eyes to verify that.

### CODE REVIEW

**Comment by cah292** [comment]
retest this please

**Comment by jhulslander** [comment]
retest this please

**Comment by jhulslander** [comment]
as for updating the oracle version, I tried updating to 12.2.0.1 and ran into some problems, so maybe it makes sense to do that in a separate story if you think that is oK

**Comment by cah292** [comment]
Yes, I'm fine with waiting for a future release to deal with the Oracle JAR upgrade, as noted earlier.

**Comment by jhulslander** [comment]
retest this please

File: src/main/resources/edu/cornell/kfs/module/purap/document/validation/configuration/PurapValidatorDefinitions.xml [hunk 1]
```diff
@@ -67,33 +66,31 @@
...
+		class="org.kuali.kfs.module.purap.document.validation.impl.PurchasingAddItemValidation" abstract="true"
+		p:businessObjectDictionaryService-ref="businessObjectDictionaryService"
+		p:dictionaryValidationService-ref="dictionaryValidationService"
+		p:businessObjectService-ref="businessObjectService"/>
 	
-	<bean id="Purchasing-importItemValidation" class="org.kuali.kfs.module.purap.document.validation.impl.PurchasingImportItemValidation" abstract="true">
-		<property name="dataDictionaryService" ref="dataDictionaryService" />
-		<property name="dictionaryValidationService" ref="dictionaryValidationService" />		
-		<property name="businessObjectService" ref="businessObjectService" />		
-		<property name="commodityCodeValidation" ref="Purchasing-commodityCodeValidation" />
```
Comment (cah292) [inline src/main/resources/edu/cornell/kfs/module/purap/document/validation/configuration/PurapValidatorDefinitions.xml:86]: This commodityCodeValidation reference should be copied over to the revised bean.

Also, I noticed that our overlay of PurchasingImportItemValidation supports some additional properties that do not appear to be in active use (and are not set by the old or new version of the bean above). We should create a user story to clean up those properties from our overlay accordingly.

File: src/main/java/edu/cornell/kfs/module/ld/document/CuSalaryExpenseTransferDocument.java [hunk 2]
```diff
@@ -50,8 +50,8 @@ public boolean generateLaborLedgerPendingEntries(AccountingLine accountingLine,
     public boolean generateLaborLedgerBenefitClearingPendingEntries(GeneralLedgerPendingEntrySequenceHelper sequenceHelper) {
         LOG.debug("started generateLaborLedgerBenefitClearingPendingEntries()");
 
-        String chartOfAccountsCode = SpringContext.getBean(ParameterService.class).getParameterValueAsString(SalaryExpenseTransferDocument.class, LaborConstants.SalaryExpenseTransfer.BENEFIT_CLEARING_CHART_PARM_NM);
-        String accountNumber = SpringContext.getBean(ParameterService.class).getParameterValueAsString(SalaryExpenseTransferDocument.class, LaborConstants.SalaryExpenseTransfer.BENEFIT_CLEARING_ACCOUNT_PARM_NM);
+        String chartOfAccountsCode = SpringContext.getBean(ParameterService.class).getParameterValueAsString(SalaryExpenseTransferDocument.class, LaborParameterConstants.BENEFIT_CLEARING_CHART);
+        String accountNumber = SpringContext.getBean(ParameterService.class).getParameterValueAsString(SalaryExpenseTransferDocument.class, LaborParameterConstants.BENEFIT_CLEARING_CHART);
```
Comment (cah292) [inline src/main/java/edu/cornell/kfs/module/ld/document/CuSalaryExpenseTransferDocument.java:54]: Should this reference BENEFIT_CLEARING_ACCOUNT instead?

File: src/main/java/org/kuali/kfs/module/purap/document/validation/impl/PurchasingAddItemValidation.java [hunk 3]
```diff
@@ -85,9 +85,8 @@ public boolean validateItemUnitPrice(PurApItem item) {
         if (item.getItemType().isLineItemIndicator()) {
             if (ObjectUtils.isNull(item.getItemUnitPrice())) {
                 valid = false;
-                String attributeLabel = dataDictionaryService.
-                                        getDataDictionary().getBusinessObjectEntry(item.getClass().getName()).
-                                        getAttributeDefinition(PurapPropertyConstants.ITEM_UNIT_PRICE).getLabel();
+                String attributeLabel = businessObjectDictionaryService.getBusinessObjectEntry(item.getClass().getName())
+                        .getAttributeDefinition(PurapPropertyConstants.ITEM_DESCRIPTION).getLabel();
```
[the remark this answers is not in the document]
Comment (jhulslander) [reply src/main/java/org/kuali/kfs/module/purap/document/validation/impl/PurchasingAddItemValidation.java:89]: good catch, I shall switch

File: src/main/resources/edu/cornell/kfs/module/purap/document/validation/configuration/PurapValidatorDefinitions.xml [hunk 4]
```diff
@@ -46,16 +46,15 @@
 
     <bean id="Purchasing-approveItemValidation" class="edu.cornell.kfs.module.purap.document.validation.impl.PurchasingProcessItemValidation" abstract="true" />
     
-    <bean id="Purchasing-commodityCodeValidation" parent="Purchasing-commodityCodeValidation-parentBean" scope="prototype" />
-	<bean id="Purchasing-commodityCodeValidation-parentBean" class="org.kuali.kfs.module.purap.document.validation.impl.PurchasingCommodityCodeValidation" abstract="true">
-		<property name="dataDictionaryService" ref="dataDictionaryService" />
-		<property name="businessObjectService" ref="businessObjectService" />	
-		<property name="parameterService" ref="parameterService" />
```
[the remark this answers is not in the document]
Comment (jhulslander) [reply src/main/resources/edu/cornell/kfs/module/purap/document/validation/configuration/PurapValidatorDefinitions.xml:53]: ok

File: src/main/resources/edu/cornell/kfs/module/purap/document/validation/configuration/PurapValidatorDefinitions.xml [hunk 5]
```diff
@@ -67,33 +66,31 @@
...
+		<property name="businessObjectDictionaryService" ref="businessObjectDictionaryService" />
 		<property name="businessObjectService" ref="businessObjectService" />	
 		<property name="parameterService" ref="parameterService" />	
 	</bean>
 	
-	<bean id="Purchasing-addItemValidation" class="org.kuali.kfs.module.purap.document.validation.impl.PurchasingAddItemValidation" abstract="true">
-		<property name="dataDictionaryService" ref="dataDictionaryService" />
-		<property name="dictionaryValidationService" ref="dictionaryValidationService" />		
-		<property name="businessObjectService" ref="businessObjectService" />		
-		<property name="commodityCodeValidation" ref="Purchasing-commodityCodeValidation" />
```
[the remark this answers is not in the document]
Comment (jhulslander) [reply src/main/resources/edu/cornell/kfs/module/purap/document/validation/configuration/PurapValidatorDefinitions.xml:79]: ok

File: src/main/java/edu/cornell/kfs/module/ld/document/CuSalaryExpenseTransferDocument.java (hunk 2 above)
Comment (jhulslander) [reply src/main/java/edu/cornell/kfs/module/ld/document/CuSalaryExpenseTransferDocument.java:54]: Yes, I shall change

**Comment by jhulslander** [comment]
I've made the code review changes suggested.  However VENDOR_HEADER_PREFIX and VENDOR_ADDRESS_STATE_CODE are still used in our code base.  I shall move those constants to CUVendorPropertyConstants

File: src/main/resources/edu/cornell/kfs/module/purap/document/validation/configuration/PurapValidatorDefinitions.xml (hunk 1 above)
Comment (jhulslander) [reply src/main/resources/edu/cornell/kfs/module/purap/document/validation/configuration/PurapValidatorDefinitions.xml:86]: I fixed the reference, and created KFSPTS-18645.

File: src/main/java/org/kuali/kfs/module/ld/document/SalaryExpenseTransferDocument.java [hunk 6]
```diff
@@ -109,10 +107,10 @@ public boolean generateLaborLedgerBenefitClearingPendingEntries(
 
         String chartOfAccountsCode = SpringContext.getBean(ParameterService.class).getParameterValueAsString(
                 SalaryExpenseTransferDocument.class,
-                LaborConstants.SalaryExpenseTransfer.BENEFIT_CLEARING_CHART_PARM_NM);
+                LaborParameterConstants.BENEFIT_CLEARING_CHART);
         String accountNumber = SpringContext.getBean(ParameterService.class).getParameterValueAsString(
                 SalaryExpenseTransferDocument.class,
-                LaborConstants.SalaryExpenseTransfer.BENEFIT_CLEARING_ACCOUNT_PARM_NM);
+                LaborParameterConstants.BENEFIT_CLEARING_CHART);
```
[the remark this answers is not in the document]
Comment (jhulslander) [reply src/main/java/org/kuali/kfs/module/ld/document/SalaryExpenseTransferDocument.java:113]: OK, will change

**Comment by jhulslander** [comment]
I believe I have made the changes requested, I am in the processing of doing some local validation so please hold off merging for now

**Comment by jhulslander** [comment]
I was able to complete local validation, KFS loads up, and I was able to route a document to final.  If you feel this PR is ready, I think it can be merged.

**Pull request merged**